src/Sha256.mo

```diff
@@ -211,6 +223,55 @@ module {
       // skipping here: i_msg := 0;
     };
 
+    public func share() : StaticSha256 = {
+      i_msg;
+      i_block;
+      high;
+      word;
+      msg = Array.freeze(msg);
+      digest = Array.freeze(digest);
+
+      // state variables in Nat16 form
+      sh = [s0h, s1h, s2h, s3h, s4h, s5h, s6h, s7h];
+      sl = [s0l, s1l, s2l, s3l, s4l, s5l, s6l, s7l];
+    };
+
+    public func unshare(state : StaticSha256) {
+      assert msg.size() == state.msg.size();
+      assert digest.size() == state.digest.size();
+
+      i_msg := state.i_msg;
+      i_block := state.i_block;
+      high := state.high;
+      word := state.word;
+
+      for (i in msg.keys()) {
+        msg[i] := state.msg[i];
+      };
+
+      for (i in digest.keys()) {
+        digest[i] := state.digest[i];
+      };
+
+      s0h := state.sh[0];
+      s0l := state.sl[0];
+      s1h := state.sh[1];
+      s1l := state.sl[1];
+      s2h := state.sh[2];
+      s2l := state.sl[2];
+      s3h := state.sh[3];
+      s3l := state.sl[3];
+      s4h := state.sh[4];
+      s4l := state.sl[4];
+      s5h := state.sh[5];
+      s5l := state.sl[5];
+      s6h := state.sh[6];
+      s6l := state.sl[6];
+      s7h := state.sh[7];
+      s7l := state.sl[7];
```

Comment:
Easier to read if we group all `sl` together and all `sh` together.

Reply:
Done